ubuntu-server-netboot.py

```diff
@@ -228,6 +241,10 @@ def setup_autoinstall(apache_dir="/var/www/html"):
         "--autoinstall-url",
         help="URL to Autoinstall config file to be used during Subiquity installation",
     )
+    parser.add_argument(
+        "--deploy-dir",
+        help="Deploy the stating dir to the target tftp dir. sudo permission is needed."
```

Comment:
I'm not really sure how this differs from `--out-dir`. Also note that sudo permission isn't required - you just need permission to write to the target dir.

Reply:
`--out-dir` is used to output `ubuntu-installer` somewhere, and `--deploy-dir` is used to move files to tftp root folder (and apache server if autoinstall is enabled).


What I want to achieve is a "one line command" to setup a netboot server, so

1. I hope the tool could copy files to `/srv/tftp` directly, and it needs the directory permission (via sudo).
2. `--out-dir` has a default sub-directory `ubuntu-installer`, and `grubnetaa64.efi` does not support the `ubuntu-installer` structure now if I understand correctly.


I may be too obsessed with the concept of "one line command to deploy". The support of `--deploy-dir` is also an individual commit. Maybe I could remove it in this pull request, and let us focus on `--autoinstall-url` better.

Reply:
I think `--autoinstall-url` would be easy to understand and useful on its own, so I'd +1 tackling that first. TBH, I'm less sure about the maintainability of dealing with the preseed/http setup ourselves. Providing example preseeds seems like a good idea, but I'm not sure how much further we can go while still doing it well.

Reply:
+1 on dropping the maintainability of dealing with the preseed/http setup ourselves. I will update the pr later on.